library/src/howMuch.ts

```diff
@@ -1,20 +1,20 @@
-import { isLeft } from "fp-ts/Either";
-import * as Codec from "io-ts/Codec";
-import { draw } from "io-ts/Decoder";
-import { matchSorter } from "match-sorter";
-import apple from "../things/things.json";
-import { ResultObject, Thing } from "./domain";
+import { ResultObject, Thing, Trip } from "./domain";
+import { getThingsFromQuery } from "./things";
+import { getTripFromQuery } from "./trips";
 
-const things = Codec.array(Thing.codec).decode(apple);
+export const howMuch = (query: string): ResultObject => {
+  const results: (Trip | Thing)[] = [];
 
-if (isLeft(things)) {
-  throw new Error(
-    `There was a problem decoding thing data: ${draw(things.left)}`
-  );
-}
+  const thingsResults = getThingsFromQuery(query);
+  const tripResult = getTripFromQuery(query);
```

Comment:
Not a fan of APIs like getXFrom or getXBy ... 
Since we have Typescript, why not shorter names and more explicit types ?
From a vocabulary standpoint, `getFromQuery` = `search`.
So why not `searchThings(ThingQueryString)` and `searchTrips(TripQueryString)`

If we want to overload `searchThings` or `searchTrips` in the future then we couold even keep it shorter like `things(ThingQueryString)` and `trips(TripQueryString)`.

Reply:
And why not keep `howMuch()` on top of these 2 ?

Reply:
Agreed on shorter names, I think `searchThings` and `searchTrips` are indeed very clear.

However, I wouldn't rely too much on types: whereas it's a great help when coding, then when reading code later or in a pull request, you don't see the types (except if you plan to place your cursor hover every variable 😬), so I think the names should be clear enough so you understand the code without the types.

For example, if I read something like `import { things } from 'howmuchcarbon';` I would expect `things` to be some data, like an object or an array. So I wouldn't advise we shorten that much the names. And usually starting functions with a verb helps to make it clear it's a function and not a variable.

But like I said, totally fine with `searchThings` and `searchTrips`, which are shorter and still clear, I'll do the changes 👍

Reply:
About `howMuch()` in the README, like I said in the PR description, I don't really see use-cases appart us where users would need it, so for clarity I suggested to remove it and focus on individual functions.

But I can add it back if you want, I don't have a strong opinion here.